### ISSUE TITLE
along method wrong results

### ISSUE THREAD

**Issue by iursevla** [Turfjs/turf#1362]
The method along seems to return results that are a little far from the line itself. Even the example from the turfjs page (See image bellow) are really off. It's supposed to be that way?
 ![along](https://i.imgur.com/x2Uxy6U.png)

**Comment by DenisCarriere** [comment]
🤔 Good observation! Not too sure what the issue would be, all the tests are passing 🤔

**Comment by iursevla** [comment]
I was trying to use this but i need it to be precise. Sometimes it's off by 400 meters or more.
Did it work well previously?

**Comment by iursevla** [comment]
Tested with  version 3.0.11 and version 5.1.6 and in both i got the same results.
[3.0.11](https://jsfiddle.net/rr_alves/28vt2v7c/)
[5.1.6](https://jsfiddle.net/rr_alves/3ns5r1bd/)

Basically created a line and sampled 50 points along the line with the same distance between them.

**Comment by vamshi29292** [comment]
I think the problem here is mercator projection.

**Comment by iursevla** [comment]
@vamshi29292 @DenisCarriere any solution or idea for this problem?

**Comment by iursevla** [comment]
The following code should return true but it returns false.

```
let line = turf.lineString([[14.326171874999998,-11.264612212504428],[17.9296875,-9.102096738726443],[18.10546875,-10.617418067950293]]);
let pt = turf.along(line, 1000, {units: 'meters'});
turf.booleanPointOnLine(pt, line);
```

The  point given by along isn't on the line itself.

**Comment by Skyerin** [comment]
+1 on this issue.

Just came across it on a project I'm working on whilst testing for another bug, where I had points [30, 15], [45, 15] and for latitude, I was getting random values depending on how far along I was on the line, such as 15.48.

Using @iursevla 's jsfiddle (thanks for that - made checking this easier for me!), along that line, it seems to be curving like a rainbow when going west to east like this.

Funnily enough, going north to south (just swapped the points around), it seems to be placing the markers correctly on the line.

**Comment by rowanwins** [comment]
Thanks for the detailed reports all, this isn't too complicated a module so I'll do some digging and see what I can find.

Cheers

**Comment by rowanwins** [comment]
So based on my investigations

The issue boils down to how bearing is calculated between points. `@turf/bearing` is designed to be used with a great circle, by comparison `@turf/rhumb-bearing` is designed for use with straight lines. For more info on the difference check this [helpful blog](https://www.movable-type.co.uk/scripts/latlong.html) or [this one](http://www.progonos.com/furuti/MapProj/Normal/CartProp/Rhumb/rhumb.html).

Currently we're using the `@turf/bearing` in `@turf/along` which probably wont make any visible difference at small scales (eg in a town), but when calculating along on large scales (eg a continent) then it starts making bigger differences. I think we're better off switching to `@turf/rhumb-bearing` as we're dealing with straight lines and not great circles.

**Comment by rowanwins** [comment]
This will be resolved by #1382 so stay tuned :)

**Comment by iursevla** [comment]
@rowanwins 
Nice job

**Comment by Skyerin** [comment]
@rowanwins amazing! thank you!

**Comment by Skyerin** [comment]
@rowanwins hi, I've seen that the PR for this was closed in favour of the V7 PR... is this in there or...? I couldn't see it myself but I could just be going blind!

**Comment by rowanwins** [comment]
Yep it is in there, I'm hoping to publish an alpha version this week of v7 so stay tuned

**Comment by vladshcherbin** [comment]
any news or workarounds with this one?

**Comment by tempranova** [comment]
This is still an issue -- any word?

**Comment by matths** [comment]
> I think the problem here is mercator projection.

That's true, I guess. That is why the MapBox GL example constructs an "arc" instead of a straight line in this example: https://docs.mapbox.com/mapbox-gl-js/example/animate-point-along-route/

Anyways, it would be nice to get points along a straight line as well. Would appreciate any news on that.

**Comment by smallsaucepan** [comment]
Ok. Think I have a handle on this. Briefly:

a) along is working ok
b) openstreetmap / geojson.io don't render lines in 3d as great circles so our examples look off
c) booleanPointOnLine treats lat / lng as planar coordinates so is fundamentally different from how along works 

***Along is doing ok***
From what I can see turf-along is doing what it should. It might not look like it, which we can go through in the next section.

To verify I calculated 4 lines of different lengths, each leaving the same point in the same direction. Then calculated "along" the longest line the same lengths as the shorter lines. The points match exactly. Differences of 1.0658141036401503e-14 for example.

To me that says ```along``` is giving us points that lie on the great circle segment between the two ends of the longest line 👍 

***Examples look off***
According to [this mapbox issue](https://github.com/mapbox/geojson.io/issues/821) lines are interpolated as a rhumb line on a Mercator plane, and then projected _back_ on to a 3d globe. So straight away this isn't the shortest path across the globe. It's the shortest path maintaining a consistent heading, so it's very very unlikely to match any points along a great circle that ```along``` is handing back.

We can see this with the illustration below.

![turf-along-mercator-great-circle-diff](https://github.com/Turfjs/turf/assets/5403628/809052e8-4746-4cac-8486-56deffdc8c40)

The solid black lines are the four lines I mentioned above (the first short line is hidden by "Finland"), and are all rendered by Mapbox as rhumb lines that "sag" toward the equator on a 3d globe. The dashed red line is the result when you fit a line through the points returned by ```along```. That looks a lot more like a great circle.

The disparities we see in the doc examples and reports above could be different examples of this. Just with far less pronounced arcs when projected.

***booleanPointOnLine makes planar calculations***
Now on to our sanity check of booleanPointOnLine. It's not going to work because it's based on planar calculations. Which bombs because that's not the approach we used to get the point we're checking against.

***Conclusion***
I've seen a few suggestions that the solution is to convert ```along``` to go full "rhumb line / Mercator". I strongly think we should go the other way, and fix booleanPointOnLine to assume great circles. Every projection to a flat plane (including Mercator) has distortion built in, esp as you near the poles. Sticking with great circles avoids much of that.

Anyway, sorry for the novel! Wanted to get a good summary down and perhaps some feedback if anyone has a counter point of view.
